packages/compiler-cli/src/ngtsc/typecheck/extended/checks/skip_hydration_binding/index.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {AST, TmplAstBoundAttribute, TmplAstNode, TmplAstTextAttribute} from '@angular/compiler';
+import ts from 'typescript';
+
+import {ErrorCode, ExtendedTemplateDiagnosticName} from '../../../../diagnostics';
+import {NgTemplateDiagnostic} from '../../../api';
+import {TemplateCheckFactory, TemplateCheckWithVisitor, TemplateContext} from '../../api';
+
+/**
+ * Ensures that the special attribute `ngSkipHydration` is not a binding and has no other
+ * value than `"true"` or an empty value.
+ */
+class NgSkipHydrationSpec extends TemplateCheckWithVisitor<ErrorCode.SKIP_HYDRATION_BINDING> {
+  override code = ErrorCode.SKIP_HYDRATION_BINDING as const;
+
+  override visitNode(
+      ctx: TemplateContext<ErrorCode.SKIP_HYDRATION_BINDING>,
+      component: ts.ClassDeclaration,
+      node: TmplAstNode|AST,
+      ): NgTemplateDiagnostic<ErrorCode.SKIP_HYDRATION_BINDING>[] {
+    /** Binding should always error */
+    if ((node instanceof TmplAstBoundAttribute) && node.name === 'ngSkipHydration') {
```

Comment:
I think it'd be great to extract the name of the attribute into a separate const in this file. It'd help avoid typos.
```suggestion
    if ((node instanceof TmplAstBoundAttribute) && node.name === NG_SKIP_HYDRATION_ATTR_NAME) {
```